rollouts/docs/development.md

```diff
@@ -0,0 +1,84 @@
+# Development
+
+## Changing Rollouts API
+
+If you change the API resources, in `api/.../*.go`, update the generated code and manifests by running:
+
+```sh
+make generate
+make manifests
+```
+
+Remove old CRDs and apply the new CRDs to your cluster by running:
+
+```sh
+make uninstall
+make install
+```
+
+## Components
+
+Rollouts comprises of several software components:
+
+* [api](../api): Definition of the KRM API supported by the rollouts controllers.
+* [config](../config): The generated rollouts manifests.
+* [manifests](../manifests): The generated rollouts manifests in kpt package format.
+* [clusterstore](../pkg/clusterstore): Represents a store of kubernetes clusters.
+* [packageclustermatcher](../pkg/packageclustermatcher): Matches discovered packages to known clusters.
+* [packagediscovery](../pkg/packagediscovery): Discovers config packages to be rolled out.
+* [tokenexchange](../pkg/tokenexchange): Exchanges tokens for authenticating GCP service accounts.
+* [rolloutsclient](../rolloutsclient): A client for the rollouts API.
+
+## Running Rollouts
+
+See dedicated documentation on running Rollouts:
+
+* [locally](running-locally.md)
+* on GKE (getting started guide coming soon)
+
+## Build the Container Images
+
+Build Docker images of Rollout components:
+
+```sh
+# Build Images
+make docker-build
+
+# Push Images to Docker Registry
+make docker-push
+
+# Supported make variables:
+# IMG     - image name and tag
+
+# Example:
+IMG=gcr.io/kpt-dev/rollouts-controller:v0.0.2 make docker-push
```

Comment:
Given that this is a dev guide, lets use `gcr.io/<your-project>/rollouts-controller:v0.0.x` as the image example? (to avoid accidental overwrites :))

Reply:
done, thanks!